api/v1beta2/flowcollector_types.go

```diff
@@ -0,0 +1,905 @@
+	//+kubebuilder:validation:Enum=DISABLED;PLAIN;SCRAM-SHA512
+	//+kubebuilder:default:=DISABLED
+	// Type of SASL authentication to use, or `DISABLED` if SASL is not used
+	Type SASLType `json:"type,omitempty"`
+
+	// Reference to the secret or config map containing the client ID
+	ClientIDReference FileReference `json:"clientIDReference,omitempty"`
+
+	// Reference to the secret or config map containing the client secret
+	ClientSecretReference FileReference `json:"clientSecretReference,omitempty"`
+}
+
+// `DebugConfig` allows tweaking some aspects of the internal configuration of the agent and FLP.
+// They are aimed exclusively for debugging. Users setting these values do it at their own risk.
+type DebugConfig struct {
+	// `env` allows passing custom environment variables to underlying components. Useful for passing
+	// some very concrete performance-tuning options, such as GOGC and GOMAXPROCS, that should not be
+	// publicly exposed as part of the FlowCollector descriptor, as they are only useful
+	// in edge debug or support scenarios.
+	//+optional
+	Env map[string]string `json:"env,omitempty"`
+}
+
+// Add more exporter types below
+type ExporterType string
+
+const (
+	KafkaExporter ExporterType = "KAFKA"
+	IpfixExporter ExporterType = "IPFIX"
+)
+
+// `FlowCollectorExporter` defines an additional exporter to send enriched flows to.
+type FlowCollectorExporter struct {
+	// `type` selects the type of exporters. The available options are `KAFKA` and `IPFIX`. `IPFIX` is unsupported (*).
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:="KAFKA";"IPFIX"
+	// +kubebuilder:validation:Required
+	Type ExporterType `json:"type"`
+
+	// Kafka configuration, such as the address and topic, to send enriched flows to.
+	// +optional
+	Kafka FlowCollectorKafka `json:"kafka,omitempty"`
+
+	// IPFIX configuration, such as the IP address and port to send enriched IPFIX flows to. [Unsupported (*)].
+	// +optional
+	IPFIX FlowCollectorIPFIXReceiver `json:"ipfix,omitempty"`
+}
+
+// `FlowCollectorStatus` defines the observed state of FlowCollector
+type FlowCollectorStatus struct {
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// `conditions` represent the latest available observations of an object's state
+	Conditions []metav1.Condition `json:"conditions"`
+
+	// Namespace where console plugin and flowlogs-pipeline have been deployed.
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:printcolumn:name="Agent",type="string",JSONPath=`.spec.agent.type`
+// +kubebuilder:printcolumn:name="Sampling (EBPF)",type="string",JSONPath=`.spec.agent.ebpf.sampling`
+// +kubebuilder:printcolumn:name="Deployment Model",type="string",JSONPath=`.spec.deploymentModel`
+// +kubebuilder:printcolumn:name="Status",type="string",JSONPath=".status.conditions[*].reason"
+// +kubebuilder:storageversion
```

Comment:
I missed some conversation, but are we finally switching to this new version as storage directly?

Reply:
As far as I understood this was the case if we wanted this in 1.4 release to have `v1alpha1` <-> `v1beta1` <-> `v1beta2` path. It's not the case anymore.

@jotak can you please confirm ?

Reply:
not sure I follow, but new api version can't be set as storage version in the same release this is what I did when transition from v1alpha1 to v1beta1

Reply:
No I think @msherif1234 is right we can't change the storage version now, we should rollback this change

Reply:
ok sure, I'm on it :+1:

Reply:
Moved the storage version to `v1beta1` in https://github.com/netobserv/network-observability-operator/pull/329/commits/656cb0fb4fd5982bad0d6f0110a2eb1bb4f2e898